### ISSUE TITLE
[Alpha 1.16.7.2]-NextGen Checkin-Printers on Kiosks not Recognizing Printer is there

### ISSUE THREAD

**Issue by kimlongcr** [SparkDevNetwork/Rock#6101]
### Description

When we try to run 16.7 on a kiosk that prints labels we are seeing the message:  No Printer Device Setup. This same kiosk using 16.6 works fine and prints the labels. 

### Actual Behavior

When we try to run 16.7 on a kiosk that prints labels we are seeing the message:  No Printer Device Setup.

We also navigated to the kiosk manager screen, entered password, and clicked "Reprint Label" option.  This time received an error "The kiosk does not have a printer defined".  Confirmed the settings do have a printer defined and was successful when clicking "Test Print Label" and a blank label printed successfully.

Tested on a second kiosk with the same results.

### Expected Behavior

Labels should print.

### Steps to Reproduce

1. Use current Check-In Kiosk that we use for 16.6 now in production
2. Update URL in the settings to nextgen checkin site 
3. Checkin child
4. Receive error "No Printer Device Setup"

### Issue Confirmation

- [X] Perform a search on the [Github Issues](https://github.com/issues?q=is%3Aissue+user%3ASparkDevNetwork+-repo%3ARock) to see if your bug or enhancement is already reported.
- [X] Reproduced the problem on a fresh install or on the [demo site](https://rock.rocksolidchurchdemo.com/).

### Rock Version

Rock McKinley 16.7 (1.16.7.2)

### Client Culture Setting

en-US

**Comment by JimMichael** [comment]
I am unable to dupe this on an ipad running the Rock 1.11.0 Testflight client. Printing from Client (using device printer), Server, or via Printer Override all work as expected. Am not able to test Bluetooth printing.

iPad is 5th gen running ios 17.7.1

**Comment by kimlongcr** [comment]
@JimMichael When you say "running the Rock 1.11.0 Testflight client", did we need to install a whole new checkin app/client on the iPad?

**Comment by mikedotmundy** [comment]
@nairdo can you add me to it as well?

**Comment by kimlongcr** [comment]
@nairdo FYI - I never received an invitation with my code.

**Comment by nairdo** [comment]
@kimlongcr  Thanks for letting us know.  We're having the person who added your email double check to see what might be going wrong with the addition.  

@mikedotmundy  Did you receive yours?

**Comment by mikedotmundy** [comment]
@nairdo Yes I did, and I successfully installed the app.

**Comment by kimlongcr** [comment]
@nairdo Received it - thanks!

**Comment by sparkdevnetwork-service** [comment]
@kimlongcr Hi Kim Just following up to see if the TestFlight app version (1.11.0) from the App Store resolved this issue for you.

**Comment by chead4** [comment]
@kimlongcr Hi Kim Just following up to see if the TestFlight app version (1.11.0) from the App Store resolved this issue for you.

**Comment by kimlongcr** [comment]
Yes, this is working great in production for us. We are going live with a few kiosks this weekend.

**Comment by chead4** [comment]
@kimlongcr  Thanks for getting back to me.  I'm going to close this issue.

**Comment by zackdutra** [comment]
All running iPad app 1.11.0

**Comment by chead4** [comment]
@zackdutra Hi Zack  Please attach a screenshot of the Device Detail block for the kiosk that you are experiencing the issue with.

**Comment by zackdutra** [comment]
<img width="1327" alt="Image" src="https://github.com/user-attachments/assets/c244ab02-832c-4d7b-bcaf-c9895938d0ef" />
<img width="1293" alt="Image" src="https://github.com/user-attachments/assets/9d41bd4b-420c-4bcd-86f9-46d8f993fdcd" />

**Comment by zackdutra** [comment]
We have always only used the IP override in the iPad app. We don't set much in the kiosk settings.
This worked fine on classic check in for reprints. It should be reprinting to the printer set by the IP override setting.
Right now we have one iPad set to classic check in and we're doing all our reprints on that one

**Comment by chead4** [comment]
@zackdutra Thanks for getting back to me and providing the screenshot and additional details Zack!

Here’s a quick breakdown of how this works and why some additional configuration in Rock is still needed.

When you use the **IP Override** setting in the iPad Check-in app, it directs the app to send the print job to that specific IP address. However, Rock still needs to know how to generate the label data properly before it can hand it off to the app. That’s why you’ll still need to define a **Printer** in Rock and associate it with each **Kiosk Device**.

For your scenario—assuming all the printers are the same model—you can streamline this by creating a single printer entry in Rock, something like **"Generic Client Printer**", without assigning an IP address. Then, assign all of your Kiosk Devices to use that printer.

Rock will use the settings from that printer definition (like DPI**, _which is new for Next-Gen check-in_) to generate the ZPL (label code). Since your Kiosk Devices are configured for **Client printing**, Rock will send that ZPL to the iPad app, which will then use the **IP Override** to determine where the label actually prints.

** This setting will be exposed for editing in v17's Device Details.

![Image](https://github.com/user-attachments/assets/5d0ba0fd-4e84-41a2-bb58-3d9284c5012d)

**Comment by zackdutra** [comment]
@chead4 I don't mind doing this. However, this is a breaking change with NextGen check in. I have never had a printer defined in rock for any kiosk, and we reprint labels every weekend. If this change is permanent, can we get it documented?

**Comment by zackdutra** [comment]
That’s great, thank you @chead4 ! I appreciate your work on this.

**Comment by chead4** [comment]
@zackdutra 

After some thoughtful review, we’ve made a small change to help simplify setup: when adding a Printer on the Device Detail block the Label Resolution (DPI) field will now default to 203.
 
In most cases, this means you won’t need to define your printer manually—unless you’re using a DPI other than 203. So if you're not using a higher-resolution printer, label reprints should work as expected once this update rolls out in v17.1.
 
We hope this makes things a little easier! You can find the updated documentation [here](https://community.rockrms.com/documentation/bookcontent/42/373#asimpleapproachtoprinting).
